commands/add.js

```diff
@@ -4,25 +4,26 @@ const addTask = require('../database')
 const add = (newTask) => {
   fs.readFile('./tasks.json', 'utf-8', (error, data) => {
     if (error) throw error
+    const taskArray = []
     if(data) {
       const taskArray = JSON.parse(data)
       taskArray.push(newTask)
       const tasks = JSON.stringify(taskArray)
-      fs.writeFile('./tasks.json', tasks, (error, data) => {
-        if (error) throw error
-        addTask(newTask)
-        console.log('Created task ' + taskArray.length);
-      })
+      addToFile(tasks)
+      console.log('Created task ' + taskArray.length);
     } else {
-      const taskArray = []
       taskArray.push(newTask)
       const tasks = JSON.stringify(taskArray)
-      fs.writeFile('./tasks.json', tasks, (error, data) => {
-        if (error) throw error
-        addTask(newTask)
-        console.log('Created task ' + taskArray.length);
-      })
+      addToFile(tasks)
+      console.log('Created task ' + taskArray.length);
     }
+    addTask(newTask)
+  })
+}
+
+const addToFile = (tasks) => {
+  fs.writeFile('./tasks.json', tasks, (error, data) => {
```

Comment:
Currently this `addToFile` function takes a json string. Why not change it to accept the tasks array and then do the stringification in this function. It probably makes more sense to handle that detail here.